target/scripts/startup/setup-stack.sh

```diff
@@ -1501,6 +1501,10 @@ function _setup_security_stack
     then
       cp /tmp/docker-mailserver/amavis.cf /etc/amavis/conf.d/50-user
     fi
+
+    sed -i -E \
+      "s|(log_level).*|\1 = ${AMAVIS_LOGLEVEL};|g" \
+      /etc/amavis/conf.d/49-docker-mailserver
```

Comment:
More precise:
`sed -i -E '/^\$log_level/'" s|(log_level).*|\1 = ${AMAVIS_LOGLEVEL};|g" /etc/amavis/conf.d/49-docker-mailserver`

Feel free to change it or leave it as it is, if you think that over complicate it 👍

Reply:
I don't know this syntax. Can you explain what this does and why one would want to use this over the current command? :D

Reply:
It ensures, that only lines starting with "$log_level" are subject for the substitution.

`sed '/PATTERN/ OPERATION' FILE`
`sed '/match this lines only/ s|foo|bar|g' FILE`

You can try both methods with a test file like this to see the difference:

```
# some comment about $log_level functionality

$log_level = ;
```

Without PATTERN matching, the comment line is also substituted which isn't desired.

> apostrophies seem off to me

This didn't work, that's why I chose single quotes around the PATTERN:

```
sed -i -E \
  "/^\$log_level/ s|(log_level).*|\1 = ${AMAVIS_LOGLEVEL};|g" \
  /etc/amavis/conf.d/49-docker-mailserver
```

I suspect some sed escape black magic..

Reply:
Final attempt 🤓:

-> Match only lines starting with `$log_level`
-> Replace `log_level.*` with `log_level = ${AMAVIS_LOGLEVEL};`
-> We only expect/interested in the first match (`.*` will match the rest of the line), so no **g**lobal substitution is needed.

`sed -i -E "/^\\\$log_level/ s|log_level.*|log_level = ${AMAVIS_LOGLEVEL};|" /etc/amavis/conf.d/49-docker-mailserver`
or
`sed -i -E '/^\$log_level/'" s|log_level.*|log_level = ${AMAVIS_LOGLEVEL};|" /etc/amavis/conf.d/49-docker-mailserver`

Reply:
I always knew `sed` could escalate quickly 😄 I agree with the statement that the current method is fine and easy to read (not that the latter is too important as we have some advanced Bash code anyways).

But thanks to you I learned something new - very much appreciated! Maybe I'll need this in the future:)

Reply:
What I don't like about `sed -i` in general is, that there is no easy way to detect, if e.g. a replacement took place. EXIT CODE is always 0, regardless if a substition took place or not. If a config file changes after an update or so, `sed` can silently fail to substitute something.

To avoid that, the sed substitute statement must be as tight as possible or a proper test must be written. Lot's of work if I look at the current script files 😄

Reply:
That is correct, I agree.